holidayextras/make-up
# Limit linting by git history date

#### What does this PR do? (please provide any background)

This introduces a new option `-g` or 'gitSince' that allows only files changed after a specific git history date to be linted.

#### What tests does this PR have?

Unit tests added for new filtering class.

#### How can this be tested?

Add this code to a project and then do:

    makeup -g <some date> dir1 dir2

By changing the date you should see only some of the files being checked.

#### Screenshots / Screencast
#### What gif best describes how you feel about this work?

![giphy](https://cloud.githubusercontent.com/assets/3158640/9619857/e326443c-510b-11e5-9bc0-35bbca86f9ff.gif)

---
#### Developer Definition of Done/Quality Checklist (for PR author to complete BEFORE code review):

#### Software Engineer or Developer review:

#### Software Engineer or project guru review:
